src/codeu/chat/server/Server.java

```diff
@@ -206,7 +246,125 @@ public void run() {
         timeline.scheduleIn(RELAY_REFRESH_MS, this);
       }
     });
-    
+  }
+
+  // This method will write the contents of dataList to a txt file
+  public void saveServer() {
+    timeline.scheduleIn(SAVE_SERVER_MS, new Runnable() {
+      @Override
+      public void run() {
+        try {
+          // Check if the data storage file exists, and if not, create it
+          if (!dataStorage.exists()) {
+            dataStorage.createNewFile();
+            LOG.info("Created data storage file.");
+          }
+
+          fileWriter = new FileWriter(dataStorage, true);
+          bufferedWriter = new BufferedWriter(fileWriter);
+
+          LOG.info("Writing sever content.");
+
+          for (int i = 0; i < dataList.size(); i++) {
+            // Write JSON object with appended identifer to file
+            // Ends up looking like: <Identifier>;<JSON Object> on every line
+            bufferedWriter.write(dataList.get(i));
+            bufferedWriter.newLine();
+          }
+        } catch (IOException ex) {
+          LOG.error(ex, "There was an exception while writing server content.");
+        } finally {
+          try {
+            //  Close and flush both readers
+            if (bufferedWriter != null) {
+              bufferedWriter.close();
+            }
+
+            if (fileWriter != null) {
+              fileWriter.close();
+            }
+          } catch (IOException ex) {
+            LOG.error(ex, "There was an exception while closing writers.");
+          }
+        }
+        // Clear Linked List in order to avoid rewrites
+        while (!dataList.isEmpty()) {
+          dataList.removeFirst();
+        }
+        timeline.scheduleIn(SAVE_SERVER_MS, this);
+      }
+    });
+  }
+
+  // This method will read the JSON from the txt file and restore the server
+  public void restoreServer() {
+    timeline.scheduleNow(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          String currentLine;
+          LOG.info("Restoring server content.");
+
+          if (!dataStorage.exists()) {
+            LOG.info("There is no storage file.");
+          }
+
+          fileReader = new FileReader(dataStorage);
+          bufferedReader = new BufferedReader(fileReader);
+
+          if (bufferedReader.readLine() == null) {
+            LOG.info("There is no data to restore.");
+          }
+
+          while ((currentLine = bufferedReader.readLine()) != null) {
+            // Split identifier from JSON and put both elements into an array
+            String[] lineElements = currentLine.split(";");
```

Comment:
This method is quite long. You could consider moving the switch-case into a separate method for better readability.